src/dns.h

```diff
@@ -6,8 +6,196 @@
 #include <stdlib.h>
 #include "map.h"
 
+// Constants
+// HSK_DNS_MAX_NAME includes all length bytes and final 0x00
+#define HSK_DNS_MAX_NAME 255
+// Byte arrays that include a length byte and final 0x00 (".")
+// will need to be size HSK_DNS_MAX_LABEL + 2
```

Comment:
> But because currently we don't have a way to differentiate, maybe it's easier that everything is in the name format 

This is my thinking as well. So all single labels or TLDs etc work just like full names do, i.e. they are read one label at a time starting with length until a label with length 0 is reached (the final 0x00)